### PR TITLE
Make DType::Struct a struct

### PR DESCRIPTION
**Pull request by gatesn**
This allows us to store e.g. a DType::Struct instead of just a DType when we want to ensure we have a struct dtype.

There are lots of FLUPs here related to #174

### CODE REVIEW

File: vortex-dtype/src/composite.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+use std::fmt::{Display, Formatter};
+
+use linkme::distributed_slice;
+use vortex_error::{vortex_err, VortexError};
+
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd, Hash)]
+#[cfg_attr(feature = "serde", derive(::serde::Serialize))]
+pub struct CompositeID(&'static str);
+
+impl CompositeID {
```
Comment (robert3005) [inline vortex-dtype/src/composite.rs:10]: Given pr title I am surprised to see these changes here. Can we change the pr title or make separate pr?

**Comment by gatesn** [comment]
I'm going to close this until we figure out what to do with Composite DTypes
